src/node.js

```diff
@@ -393,15 +399,22 @@
     function TickObject(c) {
       this.callback = c;
       this.domain = process.domain || null;
+      this._args = undefined;
     }
 
+    TickObject.prototype._setArgs = function _setArgs() {
+      this._args = arguments;
```

Comment:
This definitely should deopt since arguments must be materialized but I haven't checked